test/tests/functional/pbs_release_limited_res_suspend.py

```diff
@@ -997,3 +997,42 @@ def test_preempt_checkpoint_release_all(self):
         self.mom.add_config(c)
 
         self.helper_test_preempt_release_all("C")
+
+    def test_server_restart_with_suspened_job(self):
+        """
+        Test that when a job releases limited resources on a node and then
+        PBS server is restarted, the job is able to resume back on the same
+        node.
+        """
+        # Set ncpus in restrict_res_to_release_on_suspend server attribute
+        a = {ATTR_restrict_res_to_release_on_suspend: 'ncpus'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        # Set 2 ncpus available on the node
+        self.server.manager(MGR_CMD_DELETE, NODE, id="@default")
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.mom.shortname)
+        a = {ATTR_rescavail + '.ncpus': "2"}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+
+        # Submit a job which takes up all of the ncpus
+        j1 = Job(TEST_USER)
+        j1.set_attributes({ATTR_l + '.select': '1:ncpus=2'})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+
+        # make sure that job id is part of node's jobs attribute
+        node = self.server.status(NODE, id=self.mom.shortname)
+        self.assertTrue(jid1 in node[0]['jobs'])
+
+        # suspend job
+        self.server.sigjob(jobid=jid1, signal="suspend")
+
+        self.server.restart()
+
+        self.assertTrue(self.server.isUp())
+        self.server.expect(NODE, {'state': 'free'}, id=self.mom.shortname)
+        self.server.expect(NODE, 'jobs', op=UNSET, id=self.mom.shortname)
+
+        # resume job
+        self.server.sigjob(jobid=jid1, signal="resume")
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
```

Comment:
Since you check that the job is not in the jobs line above, you should check that it is in the jobs line here.